### PR TITLE
Fixing span manipulation and indentation of the suggestion introduced by #126187

### PR DESCRIPTION
**Pull request by surechen**
According to comments:
https://github.com/rust-lang/rust/pull/128084#issuecomment-2295254576 
https://github.com/rust-lang/rust/pull/126187/files#r1634897691

### CODE REVIEW

**Comment by cjgillot** [comment]
`no_brace_span.shrink_to_hi()` is `tcx.source_map().end_point(b.span).shrink_to_hi()`.
And you can use `tcx.source_map().indentation_before(..)` to get the correct number of spaces.

**Comment by surechen** [comment]
> `no_brace_span.shrink_to_hi()` is `tcx.source_map().end_point(b.span).shrink_to_hi()`. And you can use `tcx.source_map().indentation_before(..)` to get the correct number of spaces.



> `no_brace_span.shrink_to_hi()` is `tcx.source_map().end_point(b.span).shrink_to_hi()`. And you can use `tcx.source_map().indentation_before(..)` to get the correct number of spaces.

Done. Thank you.

**Comment by surechen** [comment]
> `no_brace_span.shrink_to_hi()` is `tcx.source_map().end_point(b.span).shrink_to_hi()`. And you can use `tcx.source_map().indentation_before(..)` to get the correct number of spaces.

Done. Thank you.

File: compiler/rustc_trait_selection/src/error_reporting/traits/suggestions.rs [hunk 1]
```diff
@@ -4668,9 +4668,8 @@ impl<'a, 'tcx> TypeErrCtxt<'a, 'tcx> {
                 && b.expr.is_none()
             {
                 sugg_spans.push((
-                    // The span will point to the closing curly brace `}` of the block.
-                    b.span.shrink_to_hi().with_lo(b.span.hi() - BytePos(1)),
-                    "\n    Ok(())\n}".to_string(),
+                    self.tcx.sess.source_map().end_point(b.span).shrink_to_lo(),
+                    "\n    Ok(())\n".to_string(),
```
Comment (cjgillot) [inline compiler/rustc_trait_selection/src/error_reporting/traits/suggestions.rs:4672]: Could you fix the indentation level too? 
And maybe avoid creating an empty line before the Ok?

Comment (surechen) [reply compiler/rustc_trait_selection/src/error_reporting/traits/suggestions.rs]: > Could you fix the indentation level too? And maybe avoid creating an empty line before the Ok?

Done. Thank you.

File: compiler/rustc_trait_selection/src/error_reporting/traits/suggestions.rs [hunk 2]
```diff
@@ -4667,10 +4667,15 @@ impl<'a, 'tcx> TypeErrCtxt<'a, 'tcx> {
...
+                // The span of '}' in the end of block.
+                let span = self.tcx.sess.source_map().end_point(b.span);
                 sugg_spans.push((
-                    // The span will point to the closing curly brace `}` of the block.
-                    b.span.shrink_to_hi().with_lo(b.span.hi() - BytePos(1)),
-                    "\n    Ok(())\n}".to_string(),
+                    span.shrink_to_lo(),
+                    format!(
+                        "{}{}",
+                        "    Ok(())\n",
```
Comment (cjgillot) [inline compiler/rustc_trait_selection/src/error_reporting/traits/suggestions.rs:4676]:
```suggestion
                        "    Ok(())\n{}",
```

**Pull request merged**